BEIMA.Backend/UserFunctions/UpdateUser.cs

```diff
@@ -67,12 +68,20 @@ public static class UpdateUser
                 }
                 else
                 {
-                    // TODO: Add validation for password length and special characters
+                    // Verify password meets all requirements
+                    if (string.IsNullOrEmpty(updatedUserRecord.Password) ||
+                       updatedUserRecord.Password.Length < 8 ||
+                       !updatedUserRecord.Password.Any(c => char.IsDigit(c)) ||
+                       !updatedUserRecord.Password.Any(c => char.IsUpper(c)) ||
```

Comment:
Also need to verify that the password also contains lower case characters as well, otherwise a user could use an all upper case password

Reply:
Maybe we should use a regex instead of LINQ expressions?

Reply:
Maybe something like this:

`^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9]).{8,}$`

Reply:
Actually, ignore the above. The above regex does not require a special character. Try this instead:

`^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$`

Reply:
Should we also include other characters like colons and slashes and stuff?

Reply:
We can go either way with it, what are your thoughts on it?